build.gradle

```diff
@@ -163,17 +164,15 @@ tasks.register("configureArtifactInfo") {
         version = "$version-$versionQualifier"
     }
 
-    def isReleaseBuild = System.getenv('RELEASE') == "1" || versionQualifier
-    String apiResponse = artifactVersionsApi.toURL().text
-
-    def dlVersions = new JsonSlurper().parseText(apiResponse)
-    String qualifiedVersion = dlVersions['versions'].grep(isReleaseBuild ? ~/^${version}$/ : ~/^${version}-SNAPSHOT/)[0]
-    if (qualifiedVersion == null) {
-        throw new GradleException("could not find the current artifact from the artifact-api ${artifactVersionsApi} for ${version}")
+    def versionSelector = new StackVersionSelector(artifactVersionsApi)
+    String qualifiedVersion = versionSelector.selectClosestVersion(version)
+    if (qualifiedVersion != version) {
+        println "WARN version $version not yet published, swith automatically to $qualifiedVersion"
```

Comment:
There's a typo: **swith**

Also, what about this instead:

"WARN version $version does not yet exist or has not been published, switching to $qualifiedVersion "

Reply:
Thanks @roaksoax I like you wording, I go for it